packages/@react-spectrum/datepicker/src/Input.tsx

```diff
@@ -121,10 +122,12 @@ function Input(props, ref) {
     validationIcon = <Checkmark data-testid="valid-icon" UNSAFE_className={iconClass} />;
   }
 
+  let modality: string = useInteractionModality();
+
   return (
     <div role="presentation" {...mergeProps(fieldProps, focusProps)} className={textfieldClass} style={style}>
       <div role="presentation" className={inputClass}>
-        <div role="presentation" className={classNames(datepickerStyles, 'react-spectrum-Datepicker-inputContents')} tabIndex={-1} ref={mergeRefs(ref, inputRef)}>
+        <div role="presentation" className={classNames(datepickerStyles, 'react-spectrum-Datepicker-inputContents')} tabIndex={modality !== 'virtual' ? -1 : undefined} ref={mergeRefs(ref, inputRef)}>
```

Comment:
Looks like modality is actually null until a interaction happens so we will still have a tabindex if this is the first element TalkBack interacts with. This seems quite unlikely so I'm fine with the current state